specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json

```diff
@@ -13221,6 +16065,111 @@
         }
       }
     },
+    "FunctionEnvelope": {
+      "description": "Web Job Information.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "FunctionEnvelope resource specific properties",
+          "properties": {
+            "name": {
+              "description": "Function name.",
+              "type": "string"
+            },
+            "functionAppId": {
+              "description": "Function App ID.",
+              "type": "string"
+            },
+            "scriptRootPathHref": {
+              "description": "Script root path URI.",
+              "type": "string"
+            },
+            "scriptHref": {
+              "description": "Script URI.",
+              "type": "string"
+            },
+            "configHref": {
+              "description": "Config URI.",
+              "type": "string"
+            },
+            "secretsFileHref": {
+              "description": "Secrets file URI.",
+              "type": "string"
+            },
+            "href": {
+              "description": "Function URI.",
+              "type": "string"
+            },
+            "config": {
+              "$ref": "#/definitions/Object",
+              "description": "Config information."
+            },
+            "files": {
```

Comment:
For "config": from the [Kudu wiki](https://github.com/projectkudu/kudu/wiki/Functions-API#creating-or-updating-a-function), it looks like they're passed in-line in the JSON passed to PUT.

For "files", it's even scarier: looks like a dictionary of filename : file-contents elements.  e.g.:
`    "files": {`
`        "foo.bat" : "@echo off \r\necho I am a batch file!\r\necho Done!\r\n"`
`    }`